### ISSUE TITLE
No way to stub private module functions with test double

### ISSUE THREAD

**Issue by kapalkat** [testdouble/testdouble.js#476]
### Description
No way to stub private module functions. 
At least I think so as I haven't found any way to do it in the docs.

### Issue

I am unit testing a module. Lets say it's moduleA.js

```js
function saySecret() {
  // Some complicated and possibly side effects
  // operations which are
  // not important for sayHi() test
  return 'Few possible options which are not important';
}

function sayHi(password) {
  let sayHiReturnedValue = 'some value';
  if (password) {
    sayHiReturnedValue += ` ${saySecret()}`; // 👈 Calling the non-export function
  }
  return sayHiReturnedValue;
}

module.exports = { sayHi };

```
Now the unit itself:
```js
const { expect } = require('chai');
const td = require('testdouble');

const anything = td.matchers.anything();

describe('sayHi()', function () {
  beforeEach(() => {
    td.reset();
    delete require.cache[require.resolve('../../moduleA')];

    this.moduleA = require('../../moduleA');
  });

  it('Should return expected value', () => {
    // ❗ What I would like to achieve: ❗
    const saySecret = td.replace(`somehow replace saySecret()`); // 👈
    td.when(saySecret(anything)).thenReturn('controlled value'); // 👈

    const testedValue = this.moduleA.sayHi('something');
    expect(testedValue).to.equal('some value controlled value');
  });
});

```

I have found some other lib which makes it possible but installing that is not an option as I would need to get unlimited amount of permissions. The lib is called `rewire` and all the steps how to do it are described [here](https://www.samanthaming.com/journal/2-testing-non-exported-functions/#non-exported-function-must-be-called-in-exported-function).
I just want to know if testdouble will give me that option. And if so how to achieve that.

### Environment

- [x] `node -v` output: v14.18.1
- [x] `npm -v` (or `yarn --version`) output: 6.14.15
- [x] `npm ls testdouble` (or `yarn list testdouble`) version: testdouble@3.16.1

**Comment by searls** [comment]
This is by design. The goal of testdouble.js is to facilitate test-driven development of a given unit by faking out that unit's dependencies via public interfaces.

In [this presentation](https://blog.testdouble.com/talks/2018-03-06-please-dont-mock-me/) I talk a bit about why mocking private APIs is discouraged.

**Comment by kapalkat** [comment]
Thanks for the quick response. I totally understand your point...unfortunately sometimes we are blocked by legacy design and changing it would cause huge problems. 
It would be just nice to have that option but it would probably create a temptation to use it against best practices...
